rollbar-java/src/main/java/com/rollbar/notifier/sender/json/JsonSerializerImpl.java

```diff
@@ -0,0 +1,191 @@
+package com.rollbar.notifier.sender.json;
+
+import static java.util.regex.Pattern.compile;
+
+import com.rollbar.api.json.JsonSerializable;
+import com.rollbar.api.payload.Payload;
+import com.rollbar.notifier.sender.result.Result;
+import com.rollbar.notifier.sender.result.ResultCode;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Implementation of the {@link JsonSerializer json serializer}.
+ */
+public class JsonSerializerImpl implements JsonSerializer {
+
+  private static final Pattern MESSAGE_PATTERN = compile("\"message\"\\s*:\\s*\"([^\"]*)\"");
+
+  private static final Pattern UUID_PATTERN = compile("\"uuid\"\\s*:\\s*\"([^\"]*)\"");
+
+  private final boolean prettyPrint;
+
+  /**
+   * Constructor.
+   */
+  public JsonSerializerImpl() {
+    this(false);
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param prettyPrint flag to pretty print the json.
+   */
+  public JsonSerializerImpl(boolean prettyPrint) {
+    this.prettyPrint = prettyPrint;
+  }
+
+  private static void serializeNumber(StringBuilder builder, Number value) {
+    builder.append(value);
+  }
+
+  private static void serializeBoolean(StringBuilder builder, Boolean value) {
+    builder.append(value ? "true" : "false");
+  }
+
+  private static void serializeNull(StringBuilder builder) {
+    builder.append("null");
+  }
+
+  private static void serializeThrowable(StringBuilder builder, Throwable value) {
+    final StringWriter writer = new StringWriter();
+    value.printStackTrace(new PrintWriter(writer));
+    builder.append(String.format("\"%s\"", writer.toString()));
+  }
+
+  private static void serializeDefault(StringBuilder builder, Object value) {
+    builder.append(String.format("\"%s\"", value));
+  }
+
+  private static Map<String, Object> asMap(Map value) {
+    try {
+      @SuppressWarnings("unchecked")
+      Map<String, Object> obj = (Map<String, Object>) value;
+      return obj;
+    } catch (ClassCastException e) {
+      Map<String, Object> obj = new LinkedHashMap<String, Object>();
+      for (Object o : value.entrySet()) {
+        Map.Entry thisOne = (Map.Entry) o;
+        Object key = thisOne.getKey();
+        Object val = thisOne.getValue();
+        obj.put(key.toString(), val);
+      }
+      return obj;
+    }
+  }
+
+  private static void serializeString(StringBuilder builder, String str) {
```

Comment:
There was a bug a fixed in this code in the old library with this function. This doesn't work correctly. So if we do want to use this current implementation then we need to at least fix this to match what rollbar-java currently has.

Reply:
I'll take a look at the last version then and update this code. Thanks for the heads up!